tests/cmaes_test.cpp

```diff
@@ -33,26 +103,101 @@ TEST_CASE("CMAESLogisticRegressionTest", "[CMAESTest]")
+{
+  SepCMAES cmaes(0, -1, 1, 32, 200, 1e-3);
+  LogisticRegressionFunctionTest(cmaes, 0.003, 0.006, 5);
+}
+
+/**
+ * Run Vd-CMA-ES with the full selection policy on logistic regression and
+ * make sure the results are acceptable.
+ */
+TEST_CASE("VDCMAESLogisticRegressionTest", "[CMAESTest]")
+{
+  VDCMAES cmaes(0, -1, 1, 32, 200, 1e-3);
+  LogisticRegressionFunctionTest(cmaes, 0.003, 0.006, 5);
+}
+
+/**
+ * Run CMA-ES with the full selection policy on Gisette dataset and 
+ * using logistic regression algorithm
+ * 
+ * See more on dataset at https://archive.ics.uci.edu/ml/datasets/Gisette 
+ * Make sure the results are acceptable.
  */
-TEST_CASE("CMAESLogisticRegressionFMatTest", "[CMAESTest]")
+TEST_CASE("CMAESGisetteTest", "[CMAESTest]")
 {
   CMAES<> cmaes(0, -1, 1, 32, 200, 1e-3);
-  LogisticRegressionFunctionTest<arma::fmat>(cmaes, 0.01, 0.02, 5);
+  LogisticRgressionGisetteData(cmaes, 1.0, 1.0, "data/gisette.csv", 5);
 }
 
 /**
- * Run CMA-ES with the random selection policy on logistic regression and
- * make sure the results are acceptable.  Use arma::fmat.
+ * Run CMA-ES with the random selection policy on Gisette dataset and 
+ * using logistic regression algorithm
+ * 
+ * See more on dataset at https://archive.ics.uci.edu/ml/datasets/Gisette 
+ * Make sure the results are acceptable.
+ */
+TEST_CASE("ApproxCMAESGisetteTest", "[CMAESTest]")
+{
+  ApproxCMAES cmaes(0, -1, 1, 32, 200, 1e-3);
+  LogisticRgressionGisetteData(cmaes, 1.0, 1.0, "data/gisette.csv", 5);
+}
+
+/**
+ * Run Active CMA-ES with the full selection policy on Gisette dataset and 
+ * using logistic regression algorithm
+ * 
+ * See more on dataset at https://archive.ics.uci.edu/ml/datasets/Gisette
+ * Make sure the results are acceptable.
+ */
+TEST_CASE("ActiveCMAESGisetteTest", "[CMAESTest]")
+{
+  ActiveCMAES cmaes(0, -1, 1, 32, 200, 1e-3);
+  LogisticRgressionGisetteData(cmaes, 1.0, 1.0, "data/gisette.csv", 5);
+}
+
+/**
+ * Run Sep-CMA-ES with the full selection policy on Gisette dataset and 
+ * using logistic regression algorithm
+ * 
+ * See more on dataset at https://archive.ics.uci.edu/ml/datasets/Gisette 
+ * Make sure the results are acceptable.
+ */
+TEST_CASE("SepCMAESGisetteTest", "[CMAESTest]")
+{
+  SepCMAES cmaes(0, -1, 1, 32, 200, 1e-3);
+  LogisticRgressionGisetteData(cmaes, 1.0, 1.0, "data/gisette.csv", 5);
+}
+
+/**
+ * Run Vd-CMA-ES with the full selection policy on Gisette dataset and 
+ * using logistic regression algorithm
+ * 
+ * See more on dataset at https://archive.ics.uci.edu/ml/datasets/Gisette 
+ * Make sure the results are acceptable.
  */
-TEST_CASE("ApproxCMAESLogisticRegressionFMatTest", "[CMAESTest]")
+TEST_CASE("VDCMAESGisetteTest", "[CMAESTest]")
 {
-  ApproxCMAES<> cmaes(0, -1, 1, 32, 200, 1e-3);
-  LogisticRegressionFunctionTest<arma::fmat>(cmaes, 0.01, 0.02, 5);
+  VDCMAES cmaes(0, -1, 1, 32, 200, 1e-3);
+  LogisticRgressionGisetteData(cmaes, 1.0, 1.0, "data/gisette.csv", 5);
```

Comment:
I don't like the idea of including some extra optimizer specific datasets for the sake of unit testing. It's fine for benchmarking, but we should keep the repo size and test time low. So in this case I would be fine with `LogisticRegressionFunctionTest` only.

Reply:
Yes, I am too. I think I just leave the file in local.